### ISSUE TITLE
podsecuritypolicytemplateprojectbindings does not validate template exists

### ISSUE THREAD

**Issue by dramich** [rancher/rancher#22556]
When creating a podsecuritypolicytemplateprojectbindings a `podSecurityPolicyTemplateId` is referenced. If that template does not exist the binding is still created successfully but the server will error saying the template is not found but no error is returned to the user. The binding sync attempts to create the psp in the downstream cluster but is unable too because the template does not exist. There should be validation to ensure the template exists before creating the binding resource. 

This happens in `pkg/controllers/user/rbac/podsecuritypolicy/binding.go`

To repro: 
Using a valid cluster/project ID send the following payload to rancher with a template ID that does not exist in rancher:

```
{
    "name": "testing",
    "namespaceId": "default",
     "podSecurityPolicyTemplateId": "doesntexist",
    "targetProjectId": "<valid projectID>"
}
```

No error is returned but the server will error with:
```
2019/08/30 11:13:11 [ERROR] PodSecurityPolicyTemplateProjectBindingController default/dddd [PodSecurityPolicyTemplateProjectBindingsLifecycleHandler] failed with : error creating policy: error getting pspt: podSecurityPolicyTemplate.management.cattle.io "doesntexist" not found
```

**Comment by jiaqiluo** [comment]
The bug is reproduced in v2.3.1 

follow the steps from the issue to send a request, and see the error in Rancher's log 
```
2019/11/01 00:01:49 [ERROR] PodSecurityPolicyTemplateProjectBindingController c-8j9mg/psptpb-8zfcz [PodSecurityPolicyTemplateProjectBindingsLifecycleHandler] failed with : error creating policy: error getting pspt: podSecurityPolicyTemplate.management.cattle.io "abcdefg" not found
2019/11/01 00:01:49 [ERROR] PodSecurityPolicyTemplateProjectBindingController c-8j9mg/psptpb-8zfcz [PodSecurityPolicyTemplateProjectBindingsLifecycleHandler] failed with : error creating policy: error getting pspt: podSecurityPolicyTemplate.management.cattle.io "abcdefg" not found
2019/11/01 00:01:50 [ERROR] PodSecurityPolicyTemplateProjectBindingController c-8j9mg/psptpb-8zfcz [PodSecurityPolicyTemplateProjectBindingsLifecycleHandler] failed with : error creating policy: error getting pspt: podSecurityPolicyTemplate.management.cattle.io "abcdefg" not found
2019/11/01 00:01:50 [ERROR] PodSecurityPolicyTemplateProjectBindingController c-8j9mg/psptpb-8zfcz [PodSecurityPolicyTemplateProjectBindingsLifecycleHandler] failed with : error creating policy: error getting pspt: podSecurityPolicyTemplate.management.cattle.io "abcdefg" not found
```

**Comment by jiaqiluo** [comment]
Failed to validate the bug in v2.3-head 1b51a5911

follow the steps from the issue to send a request, and see the error in Rancher's log, and no error returned to user
```
2019/11/01 00:08:14 [ERROR] PodSecurityPolicyTemplateProjectBindingController c-d9trj/psptpb-kt7nl [PodSecurityPolicyTemplateProjectBindingsLifecycleHandler] failed with : error creating policy: error getting pspt: podSecurityPolicyTemplate.management.cattle.io "abcdefg" not found
2019/11/01 00:08:14 [ERROR] PodSecurityPolicyTemplateProjectBindingController c-d9trj/psptpb-kt7nl [PodSecurityPolicyTemplateProjectBindingsLifecycleHandler] failed with : error creating policy: error getting pspt: podSecurityPolicyTemplate.management.cattle.io "abcdefg" not found
```

**Comment by dramich** [comment]
@jiaqiluo Sorry for the confusion, that was put into master. It's now in 2.3 as well and will be available once the build completes.

**Comment by jiaqiluo** [comment]
The bug fix is validated in Rancher:v2.3-head da8cbdb71

if the podSecurityPolicyTemplateId is an invalid value, 422 error is returned 
`
{"baseType":"error","code":"InvalidBodyContent","message":"podSecurityPolicyTemplate not found","status":422,"type":"error"}
`
